### PR TITLE
Filtering Documentation

### PR DESCRIPTION
**Pull request by rutul25**

### CODE REVIEW

File: docs/docs/guides/receiving/filtering.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+layout: default
+title: Subscriptions with filter
+nav_order: 2
+parent: Receiving messages
+grand_parent: How to Guides
+---
+
+When you receive messages from a subscription with a filter, you only receive the messages that match the filter. Metro automatically acknowledges the messages that don't match the filter. You can filter message by their attributes.
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:9]: > Metro automatically acknowledges the messages
Let's avoid internal implementation details in the documentation. Instead, can we say "Metro filters messages that match the filtering criteria before delivery"

File: docs/docs/guides/receiving/filtering.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+# Creating subscriptions with filters
+
+To create a subscription with filter capability, filtering expression need to be passed along with the payload while creating subscriptions in `filter` field.
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:13]: nit: filter capability -> filtering capability

File: docs/docs/guides/receiving/filtering.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    "topic": "projects/my-project/topics/mytopic",
+    "ackDeadlineSeconds": 90,
+    "retryPolicy": {
+        "minimumBackoff": "5s",
+        "maximumBackoff": "30s"
+    },
+    "deadLetterPolicy": {
+        "maxDeliveryAttempts": 5
+    },
+	"filter": "<Filter Expression>"
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:27]: nit: fix formatting <indentation>

File: docs/docs/guides/receiving/filtering.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+```
+
+Filter can be used in the similar manner while creating the push subscription also.
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:31]: nit: Filter -> Filtering
nit: push subscription also -> push subscriptions as well.

File: docs/docs/guides/receiving/filtering.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+# Filtering syntax
+
+To filter messages, write an expression that operates on attributes. You can write an expression that matches the key or value of the attributes. The **attributes** identifier selects the attributes in the message.
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:35]: nit: The **attributes** identifier provides access to the attributes sent as part of a message.

File: docs/docs/guides/receiving/filtering.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+
+## Functions
+
+The `hasPrefix` function filters for attributes with values that start with a substring.
+
+```
+Example: hasPrefix(attributes.domain, "co")
+```
```
Comment (vnktram) [inline docs/docs/guides/receiving/filtering.md:85]: Can we add a few more examples of complex filters that include a combination of multiple conditions, various operators, and hasPrefix clause as well?

**Review by vnktram — CHANGES_REQUESTED** [review]

File: docs/docs/guides/receiving/filtering.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+Example: `attributes.domain = "com"` , `attributes.domain != "org"`
+```
+
+The key and value are case-sensitive and must match the attribute exactly. If a key contains characters other than hyphens, underscores, or alphanumeric characters, use a string literal.
+
+
+## Boolean operators
+
+* Boolean Operators `AND` and `OR` can be used to join multiple conditions. 
+* `NOT` Operator can be used to negate the condition
```
Comment (razorao) [inline docs/docs/guides/receiving/filtering.md:64]: nit: end with .

File: docs/docs/guides/receiving/filtering.md (hunk 6 above)
Comment (razorao) [reply docs/docs/guides/receiving/filtering.md:85]: What if out filter expression has characters like `:`? Do we need to escape them? Please document that.

**Review by razorao — CHANGES_REQUESTED** [review]

File: docs/docs/guides/receiving/filtering.md (hunk 1 above)
Comment (rutul25) [reply docs/docs/guides/receiving/filtering.md]: Done

File: docs/docs/guides/receiving/filtering.md (hunk 6 above)
Comment (rutul25) [reply docs/docs/guides/receiving/filtering.md:92]: Documented the case for characters other than alphanumerics.

File: docs/docs/guides/receiving/filtering.md (hunk 4 above)
Comment (rutul25) [reply docs/docs/guides/receiving/filtering.md]: Kept the line on a safer side to avoid any confusion as the example above is a pull subscription.

**Review by razorao — APPROVED** [review]

**Review by vnktram — APPROVED** [review]
LGTM

**Pull request merged**